editor/src/clj/editor/graphics.clj

```diff
@@ -555,15 +562,45 @@
                   attribute-byte-offset
                   renderable-datas))
 
-        texcoord-index-vol (volatile! -1)
-        page-index-vol (volatile! -1)]
```

Comment:
These volatiles were previously used to keep track of how many times we've encountered an attribute with `:semantic-type-texcoord`, or `:semantic-type-page-index`, respectively. We now track this for all `semantic-types`, along with the `:attribute-byte-offset`, in the `reduce-info` `hashmap` we propagate as the state argument to the reduction function as we `reduce` over the `attributes`.